api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -113,15 +129,19 @@ public void deleteInstitution(final String shortName) {
   @Override
   public Optional<Institution> updateInstitution(
       final String shortName, final Institution institutionToUpdate) {
-    return getDbInstitution(shortName)
+    return institutionDao
+        .findOneByShortName(shortName)
         .map(DbInstitution::getInstitutionId)
         .map(
             dbId -> {
               // create new DB object, but mark it with the original's ID to indicate that this is
               // an update
+
               final DbInstitution dbObjectToUpdate =
-                  institutionMapper.modelToDb(institutionToUpdate).setInstitutionId(dbId);
-              return institutionMapper.dbToModel(institutionDao.save(dbObjectToUpdate), this);
+                  institutionDao.save(
+                      institutionMapper.modelToDb(institutionToUpdate).setInstitutionId(dbId));
+              populateAuxTables(institutionToUpdate, dbObjectToUpdate);
+              return institutionMapper.dbToModel(dbObjectToUpdate, this);
```

Comment:
Please make a private wrapper in the service like `toModel(dbObjectToUpdate)` so readers don't see this scary passing of `this` as a second argument.

Reply:
what is scary about it?

Reply:
I'm scarred by [COM programming](https://www.cs.umd.edu/~pugh/com/). I distrust any fancy footwork with `this`.

Basically this class already has a reference to the institution mapper, so it can pass whatever is needed into that method. I don't expect methods that I call to need a reference to me in general.

Reply:
easy enough, updated